components/summary-detail/index.jsx

```diff
@@ -0,0 +1,165 @@
+/* Copyright (c) 2015-present, salesforce.com, inc. All rights reserved */
+/* Licensed under BSD 3-Clause - see LICENSE.txt or git.io/sfdc-license */
+
+// Implements the [Welcome Mat design pattern](https://lightningdesignsystem.com/components/welcome-mat/) in React.
+// Based on SLDS v2.4.0
+import React from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+// ### shortid
+// [npmjs.com/package/shortid](https://www.npmjs.com/package/shortid)
+// shortid is a short, non-sequential, url-friendly, unique id generator
+import shortid from 'shortid';
+
+import assign from 'lodash.assign';
+
+import Icon from '../icon';
+
+import { SUMMARY_DETAIL } from '../../utilities/constants';
+
+const displayName = SUMMARY_DETAIL;
+
+const propTypes = {
+	assistiveText: PropTypes.shape({
+		toggleButton: PropTypes.string,
+	}),
+	/**
+	 * CSS class names to be added to the container element. `array`, `object`, or `string` are accepted.
+	 */
+	className: PropTypes.oneOfType([
+		PropTypes.array,
+		PropTypes.object,
+		PropTypes.string,
+	]),
+	/**
+	 * HTML id for component.
+	 */
+	id: PropTypes.string,
+	/**
+	 * Title for the summary
+	 */
+	title: PropTypes.string,
+	/**
+	 * Summary in detail
+	 */
+	detail: PropTypes.string,
+	/**
+	 * Sub-title for the summary.
+	 */
+	subTitle: PropTypes.string,
+	/**
+	 * Text to be shown on badge on right of the title
+	 */
+	badgeText: PropTypes.string,
+	/**
+	 * Whether summary details are visible
+	 */
+	isOpen: PropTypes.bool,
+};
+
+const defaultProps = {
+	assistiveText: {
+		toggleButton: 'Toggle details for Summary Title',
+	},
+	isOpen: false,
+};
+
+/**
+ * Use Summary Detail to show and hide details while always showing a summary
+ */
+class SummaryDetail extends React.Component {
+	constructor(props) {
+		super(props);
+		this.state = {
+			isOpen: this.props.isOpen,
+		};
+		this.generatedId = shortid.generate();
+	}
+
+	/**
+	 * Get the Summary Detail Component's HTML id. Generate a new one if no ID present.
+	 */
+	getId() {
+		return this.props.id || this.generatedId;
+	}
+
+	render() {
+		const assistiveText = assign(
+			{},
+			defaultProps.assistiveText,
+			this.props.assistiveText
+		);
+
+		const title = (
+			<h3
+				className={classNames(`slds-text-heading_small`, `slds-truncate`, {
+					'slds-p-right_small': this.props.badgeText,
+				})}
+				title={this.props.title}
+			>
+				{this.props.title}
+			</h3>
+		);
+
+		const isNotOpen = !this.state.isOpen;
+		return (
+			<div
+				id={this.getId()}
+				className={classNames(
+					`slds-summary-detail`,
+					{ 'slds-is-open': this.state.isOpen },
+					this.props.className
+				)}
+			>
+				<button
+					type="button"
+					className="slds-button slds-button_icon slds-m-right_x-small"
+					title={assistiveText.toggleButton}
+					onClick={() => this.setState({ isOpen: isNotOpen })}
```

Comment:
You should always prefer controlled components over uncontrolled in this library, so we need to add an `onToggleDetail`.